### ISSUE TITLE
react/jsx-indent bug with component "extends" syntax 

### ISSUE THREAD

**Issue by remotealex** [yannickcr/eslint-plugin-react#1640]
I'm getting a strange indent error only when using the `extends` component syntax. Stateless components don't get this error:

`Expected indentation of 10 space characters but found 8.`

``` js
class CreateArticle extends Component {
  state = {};

  handleContextRef = contextRef => this.setState({ contextRef });

  render() {
    const { contextRef } = this.state;

    return (
      <div ref={this.handleContextRef}>
        <Shelf title="Create new article" />
        <Form                                 <--- Offending line
          size="large"
          onSubmit={e => {
            e.preventDefault();
          }}
        >
         ...
       </Form>
     </div>
    )
  }
}
```

`eslint-plugin-react` version: `7.5.1`

**Comment by ljharb** [comment]
What happens if you comment out the `handleContextRef =` line?

**Comment by ljharb** [comment]
oh, i do see indentation issue there - you need one more space before `</div>`, and thus `</Form>` - not sure if that's relevant tho.

**Comment by eaardal** [comment]
I have the same problem. Only in components that extends `React.Component`, and always where one element contains several children, which are siblings, like @alexpriceonline describes

**Comment by ljharb** [comment]
@alexpriceonline @eaardal if either of you could submit a PR with a failing test case, that'd be appreciated.

**Comment by alexpriceonline** [comment]
I'll try and get one in tomorrow!

**Comment by alexpriceonline** [comment]
I can't seem to replicate it on a clean repo :( I'll try again tomorrow!

**Comment by eaardal** [comment]
Try this repo https://github.com/eaardal/eslint-indent-demo 
I stripped away as much as I could, so the app doesn't run, but you can use `npm run eslint` to see the mentioned error, and using the eslint plugin in your editor (I use VS Code) should also show the error

**Comment by alexpriceonline** [comment]
🙏 Thanks @eaardal Confirmed the issue on my machine 👍

**Comment by eaardal** [comment]
I tried reproducing the error using create-react-app, but didn't get this error there, and the most noticeable difference was that prettier was not used. But if the issue was with prettier, shouldn't running pure eslint (`npm run eslint` in my repo) then succeed? I know prettier is set-up as a plugin to eslint in the repo, but the failing rule is clearly from this repo. 🤔

**Comment by alexpriceonline** [comment]
Yeah I also tried to re-produce in CRA (without `prettier`) and couldn't. I don't have `prettier` setup as an ESLint plugin in my prod repo (where I originally got the error)... I'll pull your repo again and have a fiddle with the way `prettier` is configured.

**Comment by eaardal** [comment]
Well if you don't use prettier and got the same problem it's likely not related? I can try to remove prettier and see if the error still persists once I get off work.

**Comment by alexpriceonline** [comment]
@eaardal we do use it, but not with ESLint. Just in sequence...

**Comment by ljharb** [comment]
If you use prettier and eslint, the order of that sequence is critical; you need to run eslint *after* prettier.

**Comment by SimeonC** [comment]
I had the same issue, my problem was I was using babel-eslint ^8 (which uses the babylon parser - babel-7).
Downgrading babel-eslint to 7.2.3 fixed this issue for me.

**Comment by alexpriceonline** [comment]
@SimeonC OMG thank-you so much! 🚀

**Comment by ljharb** [comment]
Sounds like this is an incompatibility with babel 7 (which is not released yet); I’ll leave this open until it is so we don’t forget.

**Comment by ljharb** [comment]
I'm pretty confident these issues are all resolved; please file a new issue if not.
